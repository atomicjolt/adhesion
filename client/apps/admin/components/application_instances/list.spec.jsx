import React from 'react';
import { shallow } from 'enzyme';
import List from './list';

describe('application instances list', () => {
  let result;
  let props;

  beforeEach(() => {
    props = {
      applicationInstances: [
        {
          id: 4,
          application_id: 7,
        },
      ],
      settings: {},
      application: {},
      canvasOauthURL: 'https://example.com',
      disableApplicationInstance: () => {},
      deleteApplicationInstance: () => {},
      saveApplicationInstance: () => {},
      sites: { 1: { id: 1, url: 'http://www.example.com' } },
      currentSortColumn: '',
      currentSortDirection: '',
      setSort: () => {},
    };
    result = shallow(<List {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });
});
