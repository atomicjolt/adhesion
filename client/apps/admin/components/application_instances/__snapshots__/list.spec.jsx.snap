// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application instances list matches the snapshot 1`] = `
<table
  className="c-table c-table--instances"
>
  <thead>
    <tr>
      <Sortable
        column="lti_key"
        currentColumn=""
        currentDirection=""
        setSort={[Function]}
        title="LTI KEY"
      />
      <th>
        <span>
          SETTINGS
        </span>
      </th>
      <th>
        <span>
          CONFIG XML
        </span>
      </th>
      <th>
        <span>
          ENABLED
        </span>
      </th>
      <th>
        <span>
          AUTHS
        </span>
      </th>
      <Sortable
        column="created_at"
        currentColumn=""
        currentDirection=""
        setSort={[Function]}
        title="CREATED"
      />
      <th>
        ________
      </th>
      <th>
        <span>
          REQUESTS
        </span>
      </th>
      <th>
        <span>
          LTI LAUNCHES
        </span>
      </th>
      <th>
        <span>
          USERS
        </span>
      </th>
      <th>
        <span>
          ERRORS
        </span>
      </th>
      <th />
    </tr>
  </thead>
  <tbody>
    <ListRow
      application={Object {}}
      applicationInstance={
        Object {
          "application_id": 7,
          "id": 4,
        }
      }
      canvasOauthURL="https://example.com"
      delete={[Function]}
      disable={[Function]}
      key="instance_0"
      save={[Function]}
      settings={Object {}}
      sites={
        Object {
          "1": Object {
            "id": 1,
            "url": "http://www.example.com",
          },
        }
      }
    />
  </tbody>
</table>
`;
